cdisc_rules_engine/services/dataset_json_metadata_reader.py

```diff
@@ -0,0 +1,92 @@
+import pandas as pd
+from pandas import DataFrame
+import json
+
+
+class DatasetJSONMetadataReader:
+    """
+    Responsibility of the class is to read metadata
+    from Dataset-JSON file.
+    """
+
+    def __init__(self, file_path: str):
+        self._file_path = file_path
+        self._metadata_container = None
+
+    def read(self) -> dict:
+        """
+        Extracts metadata from contents of Dataset-JSON file.
+        """
+        # 2022-11-18: For the moment, we need "variable_labels", "variable_names"
+        # later, we may need "variable_name_to_label_map" and "variable_name_to_data_type_map"
+        # and maybe (???) a few others
+        # print("dataset_json_metadata_reader: starting reading_from_file = ", self._file_path)
+        f = open(self._file_path)
+        # returns JSON object as
+        # a dictionary
+        data = json.load(f)
+        # check for "clinicalData"
+        start_data_string = "clinicalData"
+        if 'clinicalData' in data:
+            start_data_string = "clinicalData"
+        else:
+            # print("could not find clinicalData key in file - trying referenceData")
+            start_data_string = "referenceData"
+        # we need to know the OID of the "ItemGroupData"
+        # TODO: very probably, this can be much simpler ...
+        item_group_data = data[start_data_string]["itemGroupData"]
+        item_group_oids = item_group_data.keys()
+        item_group_oid = list(item_group_oids)[0]
+        dataset_name = data[start_data_string]["itemGroupData"][item_group_oid]["name"]
+        dataset_label = data[start_data_string]["itemGroupData"][item_group_oid]["label"]
+        # TODO: fill variable_names and variable_labels
+        variable_names = list()
+        variable_labels = list()
+        print("DatasetJSONMetadataReader: dataset name = ", dataset_name)
+        print("DatasetJSONMetadataReader: dataset label = ", dataset_label)
+        self._metadata_container = {
+            # "variable_labels": list(dataset.contents.Label.values),
+            "variable_labels": list,
```

Comment:
Should these 'list' values be set to something else?

Reply:
I don't think so. Currently, most rules on labels work so that there is no requirement to look them up in the define.xml, as in the case of Dataset-XML. In the JSON implementation, we added names and labels to variables so that people wanting to implement Dataset-JSON do not need to rely on (or to learn) another technology (XML) to be able to develop applications.